src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptions.cs

```diff
@@ -226,10 +242,9 @@ internal OtlpExporterOptions ApplyDefaults(OtlpExporterOptions defaultExporterOp
         return this;
     }
 
-    private static string GetUserAgentString()
+    private string GetUserAgentString()
     {
-        var assembly = typeof(OtlpExporterOptions).Assembly;
-        return $"OTel-OTLP-Exporter-Dotnet/{assembly.GetPackageVersion()}";
+        return !string.IsNullOrEmpty(this.userAgentProductIdentifier) ? $"{this.userAgentProductIdentifier} {baseUserAgent}" : baseUserAgent;
```

Comment:
Am I right that now in practice the right-hand side of this expression never executes?